lib/kafka/produce_operation.rb

```diff
@@ -1,4 +1,25 @@
 module Kafka
+  # A produce operation attempts to send all messages in a buffer to the Kafka cluster.
+  # Since topics and partitions are spread among all brokers in a cluster, this usually
+  # involves sending requests to several or all of the brokers.
+  #
+  # ## Instrumentation
+  #
+  # When executing the operation, a `append_message_set.kafka` notification will be
+  # emitted for each message set that was successfully appended to a topic partition.
```

Comment:
Nah, it's a set of messages. Nothing peculiar about that. It's also the standard Kafka nomenclature, and I don't want to diverge from that too much.